Queue.rb

```diff
@@ -1,9 +1,55 @@
+require 'stack'
 class Queue
-  require 'Stack'
-  def initialize
-    stack_in = Stack.new
-    stack_out = Stack.new
-  end
-  def add(element)
-  end
+	def initialize
+		@stack_in = Stack.new
+		@stack_out = Stack.new
+	end
+
+	def empty?
+		return true if @stack_in.is_empty?
+		false
+	end
+	def add(data)
+		@stack_in.push(data)
+	end
+
+	def size
+		counter = 0
+		until @stack_in.is_empty?
+			@stack_out.push(@stack_in.peek)
```

Comment:
here you are pushing the data but not the node. I think you want to push the node with a pop not a peek